### PR TITLE
Move table checks out of cleanliness and rewrite

### PR DESCRIPTION
**Pull request by adammw**
I threw up in my mouth a little bit when i saw all the regexes in here. This change splits these db schema tests out and actually executes the `db/schema.rb` file to get the schema, which should catch any edge cases.

### Risks
- Low: change in test only

### CODE REVIEW

File: test/integration/database_schema_test.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+
+SingleCov.not_covered!
+
+describe "database schema" do
+  attr_reader :table_definitions
+
+  before(:all) do
```
Comment (grosser) [inline test/integration/database_schema_test.rb]: I don't think that works ... not sure what the first argument does though 🤷‍♂ 

`let_all ` would work ... but if it's not too slow it does not matter either

File: test/integration/database_schema_test.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  end
+
+  it "does not have boolean limit 1 in schema since this breaks mysql" do
+    table_definitions.
+      flat_map(&:columns).
+      select { |c| c.type == :boolean && c.options[:limit] == 1 }.
+      must_be_empty
+  end
+
+  it "does not have limits too big for postgres in schema" do
```
Comment (grosser) [inline test/integration/database_schema_test.rb]:
```suggestion
  it "does not have limits too big for mysql in schema" do
```

Comment (grosser) [reply test/integration/database_schema_test.rb]: .... maybe postgres, not sure
